uwblueprint/starter-code-v2
# Upgrade to React 18

## Notion ticket link

[Update React version](https://www.notion.so/uwblueprintexecs/Update-React-Version-6e2514be4e824e9b9b1bf2f633d58972)



## Implementation description
* Bumped React up to v18.x.x in `frontend/`
* Bumped Node up to v20.11.1 (latest stable version right now) everywhere
* The rest of the changes are just fixes based on linting with the upgraded linter



## Steps to test
1. Go to the frontend directory - `cd frontend`
2. Delete node_modules - `rm -r node_modules` - if you have them (if not, ignore this step).
3. Run `yarn install`.
4. Run `yarn start`.
Expected: everything should compile and localhost should be up:
![image](https://github.com/uwblueprint/starter-code-v2/assets/82423082/c32492e3-a63b-44bf-916f-9a4e02c0326e)
![image](https://github.com/uwblueprint/starter-code-v2/assets/82423082/268dc1d0-bcb8-496a-831f-38353c5331e9)


## What should reviewers focus on?
* The frontend compiles... but not sure how well it plays w/ backend and querying...
* One change I'm slightly iffy about: I had to comment out the `// import { decamelizeKeys } from "humps";` in CreateForm.tsx and UpdateForm.tsx to dodge some linter warnings about unused vars. This seems probably fine - scroll down a bit in either file and there are other examples of whole blocks of commented out code marked with `// python {`


## Checklist
